### PR TITLE
[TRAFFIC-11913]: Support Azure in `network access-point` and `network gateway` commands

### PR DESCRIPTION
**Pull request by syoung-cflt**
Release Notes
-------------

New Features
- Support creating Azure Egress Private Link endpoints in `confluent network access-point private-link egress-endpoint create my-egress-endpoint`
- Output the associated Azure Private Endpoint ID, Private Endpoint domain, and Private Endpoint IP address in `confluent network access-point private-link egress-endpoint describe` and `confluent network access-point private-link egress-endpoint list`
- Output the associated Azure subscription in `confluent network gateway describe` and `confluent network gateway list` 

Checklist
---------
- [ ] Leave this box unchecked if features are not yet available in production

What
----

References
----------

Test & Review
-------------

### CODE REVIEW

File: internal/network/command_access_point_private_link_egress_endpoint_create.go [hunk 1]
```diff
@@ -22,12 +22,17 @@ func (c *accessPointCommand) newCreateCommand() *cobra.Command {
...
+			examples.Example{
+				Text: "Create a named Azure private link egress endpoint.",
+				Code: "confluent network access-point private-link egress-endpoint create my-egress-endpoint --cloud azure --gateway gw-123456 --service /subscriptions/0000000/resourceGroups/plsRgName/providers/Microsoft.Network/privateLinkServices/privateLinkServiceName",
+			},
 		),
 	}
 
-	cmd.Flags().String("cloud", "", "Specify the cloud provider as aws.")
-	cmd.Flags().String("service", "", "Name of an AWS VPC endpoint service.")
+	cmd.Flags().String("cloud", "", "Specify the cloud provider as aws or azure.")
```
Comment (brianstrauch) [inline internal/network/command_access_point_private_link_egress_endpoint_create.go:32]: Can you use `utils.ArrayToCommaDelimitedString()` here so the options are quoted?

File: internal/network/command_access_point_private_link_egress_endpoint_create.go [hunk 2]
```diff
@@ -22,12 +22,17 @@ func (c *accessPointCommand) newCreateCommand() *cobra.Command {
...
+	cmd.Flags().String("service", "", "Name of an AWS VPC endpoint service or ID of an Azure Private Link service.")
 	addGatewayFlag(cmd, c.AuthenticatedCLICommand)
+	cmd.Flags().String("subresource", "", "Name of an Azure Private Link subresource")
```
Comment (brianstrauch) [inline internal/network/command_access_point_private_link_egress_endpoint_create.go:35]:
```suggestion
	cmd.Flags().String("subresource", "", "Name of an Azure Private Link subresource.")
```

File: internal/network/command_access_point_private_link_egress_endpoint_create.go [hunk 3]
```diff
@@ -92,6 +102,19 @@ func (c *accessPointCommand) create(cmd *cobra.Command, args []string) error {
...
+				Kind:                         "AzureEgressPrivateLinkEndpoint",
+				PrivateLinkServiceResourceId: service,
+			},
+		}
+
+		if subresource != "" {
+			specConfig.NetworkingV1AzureEgressPrivateLinkEndpoint.PrivateLinkSubresourceName = networkingaccesspointv1.PtrString(subresource)
+		}
+
+		createEgressEndpoint.Spec.Config = specConfig
```
Comment (brianstrauch) [inline internal/network/command_access_point_private_link_egress_endpoint_create.go:117]:
```suggestion
		createEgressEndpoint.Spec.Config = &networkingaccesspointv1.NetworkingV1AccessPointSpecConfigOneOf{
			NetworkingV1AzureEgressPrivateLinkEndpoint: &networkingaccesspointv1.NetworkingV1AzureEgressPrivateLinkEndpoint{
				Kind:                         "AzureEgressPrivateLinkEndpoint",
				PrivateLinkServiceResourceId: service,
			},
		}

		if subresource != "" {
			createEgressEndpoint.Spec.Config.NetworkingV1AzureEgressPrivateLinkEndpoint.PrivateLinkSubresourceName = networkingaccesspointv1.PtrString(subresource)
		}
```

**Review by sgagniere — COMMENTED** [review]
lgtm

I'll approve it when we're ready to go and we've switched to the public sdk

File: internal/network/command_access_point_private_link_egress_endpoint_create.go (hunk 3 above)
Comment (syoung-cflt) [reply internal/network/command_access_point_private_link_egress_endpoint_create.go]: updated

File: internal/network/command_access_point_private_link_egress_endpoint_create.go (hunk 1 above)
Comment (syoung-cflt) [reply internal/network/command_access_point_private_link_egress_endpoint_create.go]: yes updated thanks

File: internal/network/command_access_point_private_link_egress_endpoint_create.go [hunk 4]
```diff
@@ -22,12 +24,17 @@ func (c *accessPointCommand) newCreateCommand() *cobra.Command {
 				Text: "Create an AWS private link egress endpoint with high availability.",
 				Code: "confluent network access-point private-link egress-endpoint create --cloud aws --gateway gw-123456 --service com.amazonaws.vpce.us-west-2.vpce-svc-00000000000000000 --high-availability",
 			},
+			examples.Example{
+				Text: "Create a named Azure private link egress endpoint.",
```
Comment (brianstrauch) [inline internal/network/command_access_point_private_link_egress_endpoint_create.go:28]:
```suggestion
				Text: `Create an Azure private link egress endpoint named "my-egress-endpoint".`,
```

Comment (brianstrauch) [reply internal/network/command_access_point_private_link_egress_endpoint_create.go:28]: I think "Azure Private Link" is a proper noun here and should be capitalized, judging from some of the other text in this PR?

**Review by brianstrauch — APPROVED** [review]

File: internal/network/command_access_point_private_link_egress_endpoint_create.go (hunk 4 above)
Comment (sgagniere) [reply internal/network/command_access_point_private_link_egress_endpoint_create.go:28]: I'll capitalize it

**Review by brianstrauch — APPROVED** [review]

**Pull request merged**